### ISSUE TITLE
mergeSemantically:true throw error traversedProperty is undefined

### ISSUE THREAD

**Issue by Somebi** [jakubpawlowicz/clean-css#875]
Version used: `"clean-css": "^4.0.2",`

Here is original for reproduction:
http://pastebin.com/raw/5PZXnG2w

Setting this option to true, causing this error:
`mergeSemantically: false`

```
root@beck:/var/local/node# node minify_css.js /var/local/fpcom/public/css/style.css /var/local/fpcom/public/css/style.min.css
/var/local/node/node_modules/clean-css/lib/optimizer/level-2/merge-media-queries.js:92
      traversedRule = traversedProperty[5];
                                       ^

TypeError: Cannot read property '5' of undefined
    at allSameRulePropertiesCanBeReordered (/var/local/node/node_modules/clean-css/lib/optimizer/level-2/merge-media-queries.js:92:40)
    at mergeMediaQueries (/var/local/node/node_modules/clean-css/lib/optimizer/level-2/merge-media-queries.js:56:36)
    at level2Optimize (/var/local/node/node_modules/clean-css/lib/optimizer/level-2/optimize.js:113:15)
    at optimize (/var/local/node/node_modules/clean-css/lib/clean.js:129:5)
    at /var/local/node/node_modules/clean-css/lib/clean.js:102:29
    at Object.callback (/var/local/node/node_modules/clean-css/lib/reader/read-sources.js:30:9)
    at doApplySourceMaps (/var/local/node/node_modules/clean-css/lib/reader/apply-source-maps.js:61:23)
    at applySourceMaps (/var/local/node/node_modules/clean-css/lib/reader/apply-source-maps.js:33:5)
    at Object.callback (/var/local/node/node_modules/clean-css/lib/reader/read-sources.js:27:12)
    at doInlineImports (/var/local/node/node_modules/clean-css/lib/reader/read-sources.js:192:25)
```

node.js script used:
```
var CleanCSS = require('clean-css');
var fs = require('fs');

var css_file=process.argv[2];
if(!css_file) {
	process.exit();
}
var to_file=process.argv[3];

var input = fs.readFileSync(css_file);
var options = {
  level: {
    1: {
      cleanupCharsets: true, // controls `@charset` moving to the front of a stylesheet; defaults to `true`
      normalizeUrls: true, // controls URL normalization; defaults to `true`
      optimizeBackground: true, // controls `background` property optimizatons; defaults to `true`
      optimizeBorderRadius: true, // controls `border-radius` property optimizatons; defaults to `true`
      optimizeFilter: true, // controls `filter` property optimizatons; defaults to `true`
      optimizeFont: true, // ontrols `font` property optimizatons; defaults to `true`
      optimizeFontWeight: true, // controls `font-weight` property optimizatons; defaults to `true`
      optimizeOutline: true, // controls `outline` property optimizatons; defaults to `true`
      removeNegativePaddings: true, // controls removing negative paddings; defaults to `true`
      removeQuotes: true, // controls removing quotes when unnecessary; defaults to `true`
      removeWhitespace: true, // controls removing unused whitespace; defaults to `true`
      replaceMultipleZeros: true, // contols removing redundant zeros; defaults to `true`
      replaceTimeUnits: true, // controls replacing time units with shorter values; defaults to `true`
      replaceZeroUnits: true, // controls replacing zero values with units; defaults to `true`
      roundingPrecision: false, // rounds pixel values to `N` decimal places; `false` disables rounding; defaults to `false`
      selectorsSortingMethod: 'standard', // denotes selector sorting method; can be `natural` or `standard`; defaults to `standard`
      specialComments: 'all', // denotes a number of /*! ... */ comments preserved; defaults to `all`
      tidyAtRules: true, // controls at-rules (e.g. `@charset`, `@import`) optimizing; defaults to `true`
      tidyBlockScopes: true, // controls block scopes (e.g. `@media`) optimizing; defaults to `true`
      tidySelectors: true, // controls selectors optimizing; defaults to `true`,
    },
    2: {
      mergeAdjacentRules: true, // controls adjacent rules merging; defaults to true
      mergeIntoShorthands: true, // controls merging properties into shorthands; defaults to true
      mergeMedia: true, // controls `@media` merging; defaults to true
      mergeNonAdjacentRules: true, // controls non-adjacent rule merging; defaults to true
      mergeSemantically: true, // controls semantic merging; defaults to false
      overrideProperties: true, // controls property overriding based on understandability; defaults to true
      reduceNonAdjacentRules: true, // controls non-adjacent rule reducing; defaults to true
      removeDuplicateFontRules: true, // controls duplicate `@font-face` removing; defaults to true
      removeDuplicateMediaBlocks: true, // controls duplicate `@media` removing; defaults to true
      removeDuplicateRules: true, // controls duplicate rules removing; defaults to true
      restructureRules: true // controls rule restructuring; defaults to false
    }    
  }
};
var output = new CleanCSS(options).minify(input);

// Output format:
// { styles: 'a{font-weight:700}',
//   stats: 
//    { efficiency: 0.09999999999999998,
//      minifiedSize: 18,
//      originalSize: 20,
//      timeSpent: 15 },
//   errors: [],
//   inlinedStylesheets: [],
//   warnings: [] }

console.log(output.stats);

if(to_file) {
	var input = fs.writeFileSync(to_file,output.styles);
}
else {
	console.log(output.styles);
}



```

**Comment by jakubpawlowicz** [comment]
Top notch bug report. 👍 I can see it.

Side note: you can skip all default options if you wish to, e.g.

```js
var options = {
  level: {
    2: {
      mergeSemantically: true, // controls semantic merging; defaults to false
      restructureRules: true // controls rule restructuring; defaults to false
    }    
  }
};
```

is equal to the above.

Not sure if readme states it clearly.

**Comment by jakubpawlowicz** [comment]
It's fixed in 4.0.3.
